src/lib/Type.hs

```diff
@@ -728,6 +728,9 @@ typeCheckOp op = case op of
     i |: TC (IntRange (IdxRepVal 0) (IdxRepVal $ fromIntegral vectorWidth))
     return $ BaseTy $ Scalar sb
   ThrowError ty -> ty|:TyKind $> ty
+  -- TODO: type check that c is a character
+  -- TODO: this should really be a 32 bit integer for unicode reasons: 8 bit is just 1 codeunit
+  CodePoint c -> return $ BaseTy $ Scalar Int8Type
```

Comment:
Presumably there is more checking i should do to make sure the `c` is a character not some other type?
I don't know how though

Reply:
What does `c |: CharTy` do?

Reply:
It checks that `c` has type `CharTy`. `(|:)` is defined [here](https://github.com/google-research/dex-lang/blob/dev/src/lib/Type.hs#L327-L331) and `CharTy` is defined [here](https://github.com/google-research/dex-lang/blob/dev/src/lib/Syntax.hs#L1353-L1354).

Reply:
Yep, now we get a better error that doesn't crash the REPL

```
>=> %codePoint 'a'
97
>=> %codePoint 1
Compiler bug!
Please report this at github.com/google-research/dex-lang/issues

assertion failure ():
Char != Int32

checking decl:
tmp:Int8 = %codePoint 1
Checking body types
Checking module:
Module (Typed)
  unevaluated decls:
  tmp:Int8 = %codePoint 1
  tmp1:Int8 = tmp
  _ans_:Int8 = tmp1

  evaluated bindings:
  ()
```